misc/specs/atrium-specs-common/src/main/kotlin/ch/tutteli/atrium/specs/integration/ThrowableAssertionsSpec.kt

```diff
@@ -153,4 +159,62 @@ abstract class ThrowableAssertionsSpec(
             }.toThrow<IllegalArgumentException>()
         }
     }
+
+    describeFun(causeFeature, cause) {
+        val causeFunctions = unifySignatures(causeFeature, cause)
+
+        context("Throwable.cause is not null") {
+            val cause = IllegalArgumentException("Hello from the Cause")
+            val throwable: Throwable =
+                RuntimeException("Outer exception message", cause)
+
+            causeFunctions.forEach { (name, causeFun) ->
+                it("$name - does not throw if the assertion holds") {
+                    expect(throwable).causeFun() { toBe(cause) }
+                    expect(throwable).cause<IllegalArgumentException> {
+                        messageContains("Cause")
+                        feature(Throwable::cause).toBe(null)
+                    }
+                }
+                it("$name - throws an AssertionError") {
+                    expect {
+                        expect(throwable).causeFun() { messageContains("WRONG message") }
+                    }.toThrow<AssertionError> {
+                        messageContains(
+                            DescriptionThrowableAssertion.OCCURRED_EXCEPTION_CAUSE.getDefault() + ": java.lang.IllegalArgumentException",
+                            VALUE.getDefault() + ": \"WRONG message\""
+                        )
+                    }
+                }
+
+                // to manage users' expectations and demonstrate the flaw caused by KT-27826, and also present in the "asA"
+                it("$name (as KT-27826 is not fixed) - doesn't throws if incorrect type is expected") {
+                    val throwableWithDifferentCauseType: Throwable =
+                        RuntimeException(
+                            "Outer exception message",
+                            NumberFormatException("Cause exception: NUMBER FORMAT")
+                        )
+                    expect(throwableWithDifferentCauseType).causeFun() { messageContains("Cause exception") }
+                }
+            }
+
+        }
+
+        context("Throwable.cause is null") {
+            val throwable: Throwable = RuntimeException("Outer exception message")
+            causeFunctions.forEach { (name, causeFun) ->
+                it("$name - throws an AssertionError") {
+                    expect {
+                        expect(throwable).causeFun() { messageContains("Hello") }
+                    }.toThrow<AssertionError> {
+                        messageContains(
+                            DescriptionThrowableAssertion.OCCURRED_EXCEPTION_CAUSE.getDefault(),
```

Comment:
check that `DescriptionThrowableAssertion.NO_EXCEPTION_OCCURRED.getDefault()` (or the new description) is also in there; this way you can be sure that your implementation is actually used